src/main/java/org/fhirvp/usecase/impl/builder/FHIRResourceBuilderUseCaseImpl.java

```diff
@@ -0,0 +1,178 @@
+import org.fhirvp.model.ForecastStatus;
+import org.fhirvp.ports.impl.fhir.context.FHIRClientConfig;
+import org.fhirvp.usecase.FHIRResourceBuilderUseCase;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.lang.String;
+import java.time.LocalDate;
+import java.util.Calendar;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class FHIRResourceBuilderUseCaseImpl implements FHIRResourceBuilderUseCase {
+
+    @Inject
+    FHIRClientConfig fhirClientConfig;
+
+    public Basic buildActiveVaccinationScheme(String patientId, String vaccinationSchemeId, String changeReason) {
+        return Basic.builder()
+                .meta(Meta.builder()
+                        .profile(Canonical.of(Constants.PROFILE_BASE_URL + "vp-active-vaccination-scheme"))
+                        .build())
+                .code(CodeableConcept.builder()
+                        .coding(Coding.builder()
+                                .code(Code.builder()
+                                        .value("ActiveVaccinationScheme")
+                                        .build())
+                                .build())
+                        .build())
+                .subject(Reference.builder()
+                        .reference(fhirClientConfig.rest().base().url() + "/Patient/" + patientId)
+                        .build())
+                .extension(Extension.builder()
+                        .url(Constants.PROFILE_BASE_URL + "vp-active-vaccination-scheme-extension")
+                        .extension(
+                                Extension.builder()
+                                        .url("vaccinationScheme")
+                                        .value(Reference.builder()
+                                                .reference(fhirClientConfig.rest().base().url() + "/Basic/" + vaccinationSchemeId)
+                                                .build())
+                                        .build(),
+                                Extension.builder()
+                                        .url("changeReason")
+                                        .value(Markdown.of(changeReason))
+                                        .build()
+                        )
+                        .build())
+                .build();
+    }
+
+    public ImmunizationRecommendation buildImmunizationRecommendation
+            (
+                    String patientId, String vaccineCode, String targetDiseaseCode, ForecastStatus forecastStatus,
+                    Optional<Calendar> dateToGive, Optional<Calendar> earliestDateToGive, Optional<Calendar> latestDateToGive,
+                    List<String> supportingImmunizationIds, boolean isDeactivated, Optional<String> fulfillingImmunizationId,
+                    String supportingPopulationRecommendationId, String recommendedVaccinationDoseId
```

Comment:
This formatting style is not readable at all for me.

Reply:
Addressed at a9b6f4d